cmake/modules/extensions.cmake

```diff
@@ -2493,39 +2497,69 @@ endfunction()
 # using `-DZEPHYR_TOOLCHAIN_VARIANT=<val>`, then the value from the cache is
 # returned.
 function(zephyr_get variable)
-  cmake_parse_arguments(GET_VAR "" "SYSBUILD" "" ${ARGN})
+  cmake_parse_arguments(GET_VAR "MERGE" "SYSBUILD" "" ${ARGN})
```

Comment:
Variable is now cleared at start of function

Reply:
minor nit: you added the clearing of the var in the `sysbuild_cache_set()` commit here: https://github.com/zephyrproject-rtos/zephyr/pull/56801/commits/a0ec825aa62a5eee60769edb5fa312153e1c1585

so if you anyway need to do a PR update, then would be nice to have placed in first commit.

Reply:
Juggling too many changes, fixed.

Reply:
> Variable is now cleared at start of function

just discovered, only cleared, this part was missed: `with the introduction of MERGE we should probably read the existing local scope var`
(the sentence should probably had explicitly stated that the var should be added again on the return list)

If you try running `https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/application_development/out_of_tree_board` you'll notice that it picks up the board from ZEPHYR_BASE, and not the oot board, that is the following is printed:
```
-- Found BOARD.dts: /projects/github/ncs/zephyr/boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts
```

instead of the expected:
```
-- Found BOARD.dts: /projects/github/ncs/zephyr/samples/application_development/out_of_tree_board/boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts
```

Note, the oot_board sample should probably be reworked to do a proper test, cause the test suffers two flaws:
- it uses a board name identical to an existing Zephyr board, meaning boards can be picked up from wrong location and that is not detected in the test.
- The application source dir is default added as a search root, meaning this line doesn't make much sense:
   https://github.com/zephyrproject-rtos/zephyr/blob/3bf7f83f213f96daa9a17cd3ddd4cff0a200bb9d/samples/application_development/out_of_tree_board/CMakeLists.txt#L6

I will take a look at this in relation to the snippets work.

Reply:
Ref: https://github.com/zephyrproject-rtos/zephyr/issues/57647